app/models/update_lock.rb

```diff
@@ -3,27 +3,60 @@ def initialize(lockable)
     @lockable = lockable
 
     if lockable && !lockable.respond_to?(:transmitted_at)
-      raise ArgumentError.new("#{lockable.class} must implement transmitted_at")
+      raise ArgumentError.new("#{lockable.class} must implement transmitted_at and payload_version")
     end
   end
 
-  def check_availability!(transmitted_at)
-    raise MissingAttributeError, "transmitted_at is mandatory" unless transmitted_at
+  def check_availability!(attributes)
     return unless lockable.present?
 
-    request_transmitted_at = Integer(transmitted_at)
-    database_transmitted_at = Integer(lockable.transmitted_at)
+    attributes = attributes.with_indifferent_access
 
-    if database_transmitted_at >= request_transmitted_at
-      error = "Tried to process request with transmitted_at #{request_transmitted_at},"
-      error += " but the latest #{lockable.class} has a newer (or equal) transmitted_at of #{lockable.transmitted_at}"
-      raise OutOfOrderTransmissionError, error
-    end
+    validate_attributes(attributes)
+
+    payload_version, transmitted_at = attributes[:payload_version], attributes[:transmitted_at]
```

Comment:
You could still use parallel assignment with `attributes.values_at(:payload_version, :transmitted_at)`